universe/ShipDesign.cpp

```diff
@@ -19,6 +19,8 @@
 #include <cfloat>
 #include <boost/filesystem/fstream.hpp>
 
+extern FO_COMMON_API const int INVALID_DESIGN_ID = -1;
```

Comment:
I am going to make a guess about the situation.
In the cases where the variable is already defined in the header (like INVALID_GAME_TURN and INVALID_OBJECT_ID), the assignment in the .cpp does not have "extern FO_COMMON_API".

In this case, I have not defined INVALID_DESIGN_ID in ShipDesign.h. (That is also true for ALL_EMPIRES, which is assigned in Universe.cpp but not defined in Universe.h.)

If the suggestion is to make the "extern FO_COMMON_API" go away from the cpp file, I will likely need to add it to the ShipDesign.h file first. Let me know if you want this to happen, or to leave it in peace.